File: algorithms/413. Arithmetic Slices/solution.py
class Solution(object):
    def numberOfArithmeticSlices(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        total = cur = 0
        for i in xrange(2, len(A)):
            if A[i] - A[i-1] == A[i-1] - A[i-2]:
                cur += 1
                total += cur
            else:
                cur = 0
        return total

File: algorithms/413. Arithmetic Slices/solution.cc
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& A) {
        int cur = 0, total = 0;
        for ( int i = 2; i < A.size(); ++i ) {
            if ( A[i] - A[i-1] == A[i-1] - A[i-2] ) {
                ++cur;
                total += cur;
            } else {
                cur = 0;
            }
        }
        return total;
    }
};
